src/token-gated-minter/TokenGatedMinter.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {IERC165} from "forge-std/interfaces/IERC165.sol";
+import {IERC721} from "forge-std/interfaces/IERC721.sol";
+import {IERC721Drop} from "zora-drops-contracts/interfaces/IERC721Drop.sol";
+
+contract TokenGatedMinter {
+    struct TokenGate {
+        uint256 mintPrice;
+        uint256 mintLimitPerToken;
+    }
+
+    // token gate contract address => token gate details
+    mapping(address => TokenGate) public tokenGates;
+
+    // token gate address => token gate tokenId => was used for minting
+    mapping(address => mapping(uint256 => bool)) public tokenUsedForMinting;
+
+    address payable public immutable mintingToken;
+
+    modifier onlyTokenAdmin() {
+        require(
+            IERC721Drop(mintingToken).isAdmin(msg.sender),
+            "TokenGatedMinter: not token admin"
+        );
+        _;
+    }
+
+    constructor(address payable _mintingToken) {
+        mintingToken = _mintingToken;
+    }
+
+    function setTokenGate(
+        address _token,
+        uint256 _mintPrice,
+        uint256 _mintLimitPerToken
+    ) external onlyTokenAdmin {
+        if (_token == address(0)) {
+            return;
+        }
+        if (_mintLimitPerToken == 0) {
+            delete tokenGates[_token];
+            return;
+        }
+        tokenGates[_token] = TokenGate({
+            mintPrice: _mintPrice,
+            mintLimitPerToken: _mintLimitPerToken
+        });
+    }
+
+    function mintWithAllowedTokens(
+        address _tokenGate,
+        uint256 _amountToMint,
```

Comment:
Can this just be `_tokenIds.length`?

Reply:
would be simpler but what if drop admin wants users to be able to mint 2+ tokens per mint pass?